okio/src/commonMain/kotlin/okio/internal/OkioMessageDigest.kt

```diff
@@ -0,0 +1,24 @@
+package okio.internal
+
+// TODO refactor to typealias after this is resolved https://youtrack.jetbrains.com/issue/KT-37316
+internal interface OkioMessageDigest {
+
+  /**
+   * Update the digest using [input]
+   */
+  fun update(input: ByteArray)
+
+  /**
+   * Complete the hash calculaion and return the hash as a [ByteArray]
+   */
+  fun digest(): ByteArray
+
+  companion object {
+    fun getInstance(algorithm: OkioMessageDigestAlgorithm): OkioMessageDigest = when (algorithm) {
+      OkioMessageDigestAlgorithm.SHA_1 -> Sha1MessageDigest()
+      else -> TODO()
+    }
+  }
+}
+
+internal expect class Sha1MessageDigest constructor(): OkioMessageDigest
```

Comment:
We shouldn't need one class per hash, at least not on the JVM.

I think the shape of this should be
```kotlin
interface OkioMessageDigest {
  fun update(..)
  fun digest(..)
}

expect fun newMessageDigest(algorithm: String): OkioMessageDigest
```
Then JVM can just delegate to `MessageDigest.getInstance` and non-JVM can just do the switch to whatever it needs.

Reply:
Love it. Will make that change.

Reply:
stubs updated.